### PR TITLE
Support gitignore-style patterns for tracked repo filtering (#374)

### LINKED ISSUE #374: Support gitignore-style patterns for tracked repo filtering in GitHub extension
## Summary

The DevDocket GitHub extension currently only allows filtering tracked repos by explicit inclusion (listing specific repo names). This is limiting for users who work across many repositories in an organization.

## Proposal

Replace the explicit repo list with a more flexible pattern-based filtering system, similar to `.gitignore` syntax. This would support:

- **Wildcards** — e.g., `myorg/*` to match all repos in an org
- **Exclusions** — e.g., `!myorg/deprecated-*` to exclude repos matching a pattern
- **Multiple lines** — each line is a separate pattern, evaluated in order
- **Negation** — prefix with `!` to exclude previously matched repos

### Example Configuration

```
# Include all repos in my org
myorg/*

# But exclude archived/deprecated ones
!myorg/deprecated-*
!myorg/archive-*

# Also include a specific repo from another org
other-org/shared-lib
```

## Benefits

- Scales better for users in large organizations
- Reduces configuration maintenance when new repos are created
- Familiar syntax for developers already used to `.gitignore`

---

### Configuration Change

**Setting:** `devdocketGithub.repos`
**Type:** Changes from `string[]` to `string` (breaking change)
**Format:** Multiline text, one pattern per line

**Parsing rules:**
- Empty lines and lines starting with `#` are ignored (comments)
- Lines starting with `!` are exclusion patterns (negation)
- All other lines are inclusion patterns
- Patterns are evaluated in order; **last match wins** (gitignore-style)
- `*` matches any characters within a single path segment (owner or repo name)
- No support for `?`, `[]`, `**`, or other extended glob features

**Empty config behavior:** Same as today — fetch all assigned items globally.

**Breaking change:** The setting type changes from `string[]` to `string`. Existing users with array-style configs will need to migrate manually. This will be documented in the changelog.

### Pattern Parser & Matcher

**New module:** `packages/github/src/repoPattern.ts`

**Data structure:**
```ts
interface RepoPattern {
  original: string;     // Raw pattern text
  negated: boolean;     // true if starts with !
  ownerGlob: string;    // Owner segment (e.g., "myorg" or "*")
  repoGlob: string;     // Repo segment (e.g., "*", "deprecated-*")
}
```

**Core functions:**

1. **`parseRepoPatterns(config: string): RepoPattern[]`** — Splits multiline string, strips comments and blank lines, parses each line into a `RepoPattern`.

2. **`matchesRepoPatterns(repo: string, patterns: RepoPattern[]): boolean`** — Evaluates an `owner/repo` string against all patterns in order. Converts globs to regex (e.g., `deprecated-*` → `/^deprecated-.*$/`). Returns whether the repo is included after applying all patterns (last match wins).

3. **`extractOwners(patterns: RepoPattern[]): string[]`** — Returns the unique set of owner names from non-negated patterns. Used to determine which orgs/users to query for repo discovery.

**Glob-to-regex conversion:** `*` → `.*`, all other characters regex-escaped. No `?`, `[]`, or `**`.

### Repo Resolver (Discovery via GitHub API)

**New module:** `packages/github/src/repoResolver.ts`

**Purpose:** Given parsed patterns, resolve wildcards to concrete `owner/repo` names using the GitHub Search API.

**Resolution flow:**

1. Parse patterns and separate into positive (inclusion) and negative (exclusion) sets
2. **If no positive patterns exist** (negation-only config):
   - Use the global fetch path (same as empty config)
   - Post-filter the globally-fetched results by running `matchesRepoPatterns()` — any item whose `owner/repo` matches a negation is excluded
3. **If positive patterns exist** (with or without negations):
   - Call `extractOwners(patterns)` to get unique owners (e.g., `["myorg", "other-org"]`)
   - For each owner with wildcard patterns, use GitHub Search API to discover repos
   - For exact patterns (no wildcards), add them directly to the candidate list (no API call needed)
   - Run `matchesRepoPatterns()` against the full candidate list to produce the final resolved repo list
   - Fetch per resolved repo using existing per-repo fetch logic

**Key function signature:**
```ts
async function resolveRepos(
  patterns: RepoPattern[],
  token: string,
  signal?: AbortSignal,
): Promise<{ repos: string[]; failures: string[] }>
```

**Error handling:** If an owner can't be fetched (404, auth error), log a warning and include it in `failures` — don't block other owners from resolving.

### Provider Integration

All three providers (`GitHubIssueProvider`, `GitHubPrReviewProvider`, `GitHubMyPrsProvider`) follow the same pattern:

1. **`getConfiguredRepos()` becomes `getConfiguredPatterns()`** — returns parsed `RepoPattern[]` from the multiline string setting

2. **Updated fetch flow:**
   - `patterns.length === 0` → use existing global fetch (no change)
   - Negation-only patterns → global fetch + post-filter results
   - Positive patterns present → resolve repos via `resolveRepos()`, then fetch per resolved repo

3. **Shared logic:** The common "resolve repos from patterns" step is extracted into `repoResolver.ts` so all three providers call the same helper

4. **Validation:** The existing `isValidGitHubRepo()` check on each repo entry is replaced by pattern parsing validation at parse time. Per-repo fetch paths still validate resolved names before making API calls.

### Testing Strategy

**Unit tests for `repoPattern.ts`:**
- Parsing: comments, blank lines, negation, wildcards, invalid patterns
- Matching: exact match, glob match, negation, evaluation order (last match wins), edge cases (empty patterns, all negations)
- `extractOwners`: deduplication, negation-only patterns excluded

**Unit tests for `repoResolver.ts`:**
- Mock GitHub API responses for repo discovery
- Exact patterns bypass API calls
- Wildcard patterns trigger discovery + filtering
- Error handling (404s, partial failures)
- Mixed patterns (some exact, some wildcard)
- Negation-only patterns use global fetch path

**Integration tests for providers:**
- Config with patterns resolves correctly and triggers per-repo fetches
- Empty config still uses global fetch path
- Negation-only config uses global fetch with post-filtering
- Failures from resolution are logged but don't crash

### PR DESCRIPTION
**Pull request by mthalman**
## Summary

Support gitignore-style patterns for tracked repo filtering in the GitHub extension.

Closes #374

### New modules
- **`repoPattern.ts`** — Parses multiline config strings into repo patterns supporting gitignore-style syntax: wildcards (`*`), exclusions (`!`), comments (`#`). Case-insensitive matching with last-match-wins semantics.
- **`repoResolver.ts`** — Resolves wildcard patterns to concrete repo lists via GitHub API (org endpoint with user fallback, paginated up to 1000 repos).

### Config change
The `devDocketGithub.repos` setting changes from `string[]` to a multiline `string` with `editPresentation: multilineText`. Example:
```
myorg/*
!myorg/archived-repo
other-owner/specific-repo
```

**Backward compatible** — the old `string[]` format is detected at runtime and handled correctly.

### Provider updates
All 3 GitHub providers (`GitHubIssueProvider`, `GitHubMyPrsProvider`, `GitHubPrReviewProvider`) now use the shared `getConfiguredPatterns()` and `resolveConfiguredRepos()` methods from `BaseGitHubProvider`.

Key behaviors:
- **Empty config** → global fetch (all assigned items)
- **Negation-only patterns** → global fetch + post-filter
- **Positive patterns resolve to nothing** → emit empty (no silent fallback to global)

### Tests
- 251-line `repoPattern.test.ts` — parsing, matching, edge cases
- `repoResolver.test.ts` — 12 tests covering exact/wildcard resolution, pagination, org/user fallback, exclusion filtering, abort handling, parallel resolution
- Backward-compat test for legacy array config format
- All existing tests updated to use new string config format

### CODE REVIEW

File: packages/github/src/repoResolver.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          Accept: 'application/vnd.github+json',
+          'User-Agent': 'DevDocket-VSCode',
+          'X-GitHub-Api-Version': '2022-11-28',
+        },
+        signal: combineSignals(signal, 30_000),
+      });
+    } catch (err) {
+      if (err instanceof Error && err.name === 'AbortError') { throw err; }
+      logger.warn(`Failed to list repos from ${nextUrl}`, err);
+      return allRepos.length > 0 ? allRepos : null;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoResolver.ts]: Fixed in 676af20 — network errors now return empty array instead of null, so the org-to-user fallback only triggers on actual 404s.

File: packages/github/src/repoResolver.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+/**
+ * Resolve repo patterns into a concrete list of "owner/repo" strings.
+ * Exact patterns are returned as-is. Wildcard patterns trigger GitHub API
+ * listing for the owner (tries org endpoint first, falls back to user).
+ */
+export async function resolveRepos(
+  patterns: RepoPattern[],
+  token: string,
+  signal?: AbortSignal,
+): Promise<string[]> {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoResolver.ts]: The PR description does not claim resolveRepos returns { repos, failures } — it describes it as resolving wildcard patterns to concrete repo lists. The function signature is Promise<string[]> which matches the description. No change needed.

File: packages/github/src/githubPrReviewProvider.ts [hunk 3]
```diff
@@ -30,30 +31,45 @@ export class GitHubPrReviewProvider extends BaseGitHubProvider {
...
 
     logger.info(`Discovered ${prs.length} PR review requests`);
 
+    // Post-filter for negation-only patterns
+    const filteredPrs = useGlobalFetch && patterns.length > 0
+      ? prs.filter(pr => {
+          const repoName = parseRepoFromUrls(pr.html_url, pr.repository_url);
+          return matchesRepoPatterns(repoName, patterns);
+        })
+      : prs;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubPrReviewProvider.ts]: Fixed in 9d5b298 — moved log after post-filter so it reflects the actual published count.

File: packages/github/src/githubProvider.ts [hunk 4]
```diff
@@ -21,10 +22,25 @@ export class GitHubIssueProvider extends BaseGitHubProvider {
...
+    // Post-filter for negation-only patterns
+    const filteredIssues = useGlobalFetch && patterns.length > 0
+      ? issues.filter(issue => {
+          const repoName = parseRepoFromUrls(issue.html_url, issue.repository_url);
+          return matchesRepoPatterns(repoName, patterns);
+        })
+      : issues;
+
+    const items: DiscoveredItem[] = filteredIssues.map((issue) => {
       const repoName = parseRepoFromUrls(issue.html_url, issue.repository_url);
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubProvider.ts]: Fixed in e9534b2 — repo name is now parsed once per issue, then reused for filtering and mapping.

File: packages/github/src/test/repoResolver.test.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+describe('resolveRepos', () => {
+  let mockFetch: ReturnType<typeof vi.fn>;
+
+  beforeEach(() => {
+    mockFetch = vi.fn();
+    vi.stubGlobal('fetch', mockFetch);
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/repoResolver.test.ts]: Fixed in 5598372 — added vi.unstubAllGlobals() to afterEach for proper test isolation.

File: packages/github/src/githubMyPrsProvider.ts [hunk 6]
```diff
@@ -76,14 +83,26 @@ export class GitHubMyPrsProvider extends BaseGitHubProvider {
     const authoredUrls = new Set(authoredResult.prs.map(pr => pr.html_url));
     const uniqueAssignedPrs = assignedResult.prs.filter(pr => !authoredUrls.has(pr.html_url));
 
-    const allPrs = [...authoredResult.prs, ...uniqueAssignedPrs];
+    // Post-filter for negation-only patterns
+    const repoFilter = (pr: GitHubIssue) => {
+      const repoName = parseRepoFromUrls(pr.html_url, pr.repository_url);
+      return matchesRepoPatterns(repoName, patterns);
+    };
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts]: Fixed in 2f59c92 — moved log after post-filtering to show accurate counts.

File: packages/github/src/githubPrReviewProvider.ts [hunk 7]
```diff
@@ -30,30 +31,45 @@ export class GitHubPrReviewProvider extends BaseGitHubProvider {
...
     const { prs, failures } = await this.fetchReviewRequestedPrs(accessToken, repos, signal);
 
-    logger.info(`Discovered ${prs.length} PR review requests`);
+    // Post-filter for negation-only patterns
+    const filteredPrs = useGlobalFetch && patterns.length > 0
+      ? prs.filter(pr => {
+          const repoName = parseRepoFromUrls(pr.html_url, pr.repository_url);
+          return matchesRepoPatterns(repoName, patterns);
+        })
+      : prs;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubPrReviewProvider.ts]: Fixed in aeed3ef — repo names are cached in a Map and reused for filtering and item mapping.

File: packages/github/src/githubMyPrsProvider.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts]: Fixed in a8e3067 — repo names cached in a Map and reused for filtering and item construction.

File: packages/github/src/repoPattern.ts [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    }
+    const rawPattern = p.pattern;
+    const slashIndex = rawPattern.indexOf('/');
+    if (slashIndex === -1) {
+      continue;
+    }
+    const owner = rawPattern.slice(0, slashIndex).trim();
+    if (owner && !owner.includes('*')) {
+      owners.add(owner);
+    }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts]: Acknowledged — wildcard owners (*/repo, */*) are intentionally unsupported as they would require listing all GitHub orgs/users, which is impractical. The current behavior (resolves to empty) is correct for now. A future enhancement could add validation during parsing to warn about unsupported patterns.

File: packages/github/package.json [hunk 9]
```diff
@@ -24,12 +24,10 @@
...
-          "type": "array",
-          "items": {
-            "type": "string"
-          },
-          "default": [],
-          "description": "GitHub repositories to watch (e.g. 'owner/repo'). Leave empty to fetch all assigned issues across all repositories."
+          "type": "string",
+          "default": "",
+          "editPresentation": "multilineText",
+          "description": "Gitignore-style patterns for GitHub repositories to watch. One pattern per line. Supports wildcards (*), exclusions (!), and comments (#). Examples:\nmyorg/*\n!myorg/archived-repo\nother-owner/specific-repo\nLeave empty to fetch all assigned items across all repositories."
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/package.json]: Intentionally keeping type: string only. The editPresentation: multilineText setting requires type: string to render properly. The runtime backward-compat handles old arrays correctly, and the squiggle will naturally disappear once users interact with the setting in the UI (which overwrites the value as a string). Adding oneOf would complicate the schema for a transient migration period.

File: packages/github/src/repoResolver.ts [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+  const orgUrl = `https://api.github.com/orgs/${encodeURIComponent(owner)}/repos?per_page=100`;
+  let repos = await fetchPaginatedRepos(orgUrl, token, signal);
+
+  if (repos === null) {
+    // Org endpoint returned 404; try user endpoint
+    const userUrl = `https://api.github.com/users/${encodeURIComponent(owner)}/repos?per_page=100`;
+    repos = await fetchPaginatedRepos(userUrl, token, signal);
+  }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoResolver.ts]: Acknowledged — this is a known GitHub API limitation. /users/{username}/repos only returns public repos. Using /user/repos for the authenticated user's own repos would require an additional API call to detect the current user. This is a valid enhancement for a follow-up issue but beyond the scope of this PR.

File: packages/github/src/repoResolver.ts [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+  token: string,
+  signal?: AbortSignal,
+): Promise<string[] | null> {
+  const allRepos: string[] = [];
+  let nextUrl: string | null = startUrl;
+  let page = 0;
+  const maxPages = 10;
+
+  while (nextUrl && page < maxPages) {
+    if (signal?.aborted) {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoResolver.ts]: Fixed in b9a5a61 — added a warning log when the pagination limit is reached and more pages remain.

File: packages/github/src/repoPattern.ts [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+ * Parse a multiline config string into an array of repo patterns.
+ * Supports gitignore-style syntax: comments (#), exclusions (!), wildcards (*).
+ */
+export function parseRepoPatterns(config: string): RepoPattern[] {
+  const lines = config.split('\n');
+  const patterns: RepoPattern[] = [];
+
+  for (const rawLine of lines) {
+    const line = rawLine.trim();
+    if (line === '' || line.startsWith('#')) {
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts]: Fixed in 597517e — added inline comment stripping and renamed isExclusion to isNegation.

File: packages/github/src/repoPattern.ts [hunk 13]
```diff
@@ -0,0 +1,64 @@
+export interface RepoPattern {
+  pattern: string;
+  isExclusion: boolean;
+  regex: RegExp;
+}
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts:5]: Fixed in 597517e — renamed to isNegation across source and tests.

File: packages/github/package.json [hunk 14]
```diff
@@ -24,12 +24,10 @@
...
-          "items": {
-            "type": "string"
-          },
-          "default": [],
-          "description": "GitHub repositories to watch (e.g. 'owner/repo'). Leave empty to fetch all assigned issues across all repositories."
+          "type": "string",
+          "default": "",
+          "editPresentation": "multilineText",
+          "description": "Filter out repositories from DevDocket results. Uses gitignore-style patterns, one per line. Matching repositories are excluded; unmatched repositories are shown.\n\nExamples:\nmyorg/old-*          # filter out repos starting with old-\ndevdocket/*          # filter out all devdocket repos\n!devdocket/devdocket # except keep this one\n\nSupports wildcards (*), negation (!), and comments (#).\nLeave empty to show all repositories."
         },
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/package.json]: The PR was refactored to remove repoResolver.ts entirely. Global fetch + post-filter is the actual approach. PR description will be updated.

File: packages/github/src/test/repoPattern.test.ts [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+    expect(patterns).toHaveLength(1);
+    expect(patterns[0].isExclusion).toBe(true);
+    expect(patterns[0].regex.test('owner/repo')).toBe(true);
+  });
+
+  it('skips comments', () => {
+    const patterns = parseRepoPatterns('# this is a comment\nowner/repo');
+    expect(patterns).toHaveLength(1);
+    expect(patterns[0].pattern).toBe('owner/repo');
+  });
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/repoPattern.test.ts:41]: Fixed in 597517e — added two tests for inline trailing comments.

File: packages/github/src/githubProvider.ts [hunk 16]
```diff
@@ -40,10 +51,8 @@ export class GitHubIssueProvider extends BaseGitHubProvider {
     logger.info(`Discovered ${items.length} GitHub issues`);
     this._onDidDiscoverItems.fire(items);
 
-    if (failures.length > 0) {
-      const message = failures.length === 1
-        ? `Failed to fetch issues from ${failures[0]}`
-        : `Failed to fetch issues from ${failures.length} repositories`;
+    if (failed) {
+      const message = 'Failed to fetch issues from all repositories';
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubProvider.ts]: Fixed in 3d674d9 — changed to 'Failed to fetch assigned issues'.

File: packages/github/src/githubPrReviewProvider.ts [hunk 17]
```diff
@@ -78,10 +90,8 @@ export class GitHubPrReviewProvider extends BaseGitHubProvider {
 
     this._onDidDiscoverItems.fire(items);
 
-    if (failures.length > 0) {
-      const message = failures.length === 1
-        ? `Failed to fetch PR review requests from ${failures[0]}`
-        : `Failed to fetch PR review requests from ${failures.length} repositories`;
+    if (failed) {
+      const message = 'Failed to fetch PR review requests from all repositories';
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubPrReviewProvider.ts]: Fixed in 96df566 — changed to 'Failed to fetch PR review requests'.

File: packages/github/src/repoPattern.ts [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+ * `!` (negation) patterns un-filter (re-include) them. Last match wins.
+ * Returns true if the repo should be kept, false if filtered out.
+ */
+export function matchesRepoPatterns(repo: string, patterns: RepoPattern[]): boolean {
+  if (patterns.length === 0) { return true; }
+  let result = true;
+  for (const p of patterns) {
+    if (p.regex.test(repo)) {
+      result = p.isNegation;
+    }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts]: Fixed in 9f40673 — expanded JSDoc to clearly document the filter-out semantics. The implementation is intentionally gitignore-style: positive patterns exclude, negation re-includes. This matches .gitignore where listed patterns are ignored (excluded) and ! un-ignores.

File: packages/github/src/githubProvider.ts [hunk 19]
```diff
@@ -21,11 +22,21 @@ export class GitHubIssueProvider extends BaseGitHubProvider {
...
   protected async fetchAndPublish(accessToken: string, isUserTriggered: boolean, signal?: AbortSignal): Promise<void> {
     logger.info('Fetching assigned issues...');
-    const repos = this.getConfiguredRepos();
-    const { issues, failures } = await this.fetchAssignedIssues(accessToken, repos, signal);
+    const patterns = this.getConfiguredPatterns();
 
-    const items: DiscoveredItem[] = issues.map((issue) => {
-      const repoName = parseRepoFromUrls(issue.html_url, issue.repository_url);
+    const { issues, failed } = await this.fetchAllAssignedIssues(accessToken, signal);
+
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubProvider.ts:28]: Fixed in 9f40673 — updated class comment to reflect global-fetch + post-filter architecture.

File: packages/github/src/repoPattern.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts]: Fixed in ad710e3 — clarified JSDoc to say 'trailing comments' instead of 'inline comments'.

File: packages/github/src/githubProvider.ts (hunk 19 above)
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubProvider.ts:28]: Fixed in 529d6c0 — removed legacy array compat since setting was renamed to filteredRepos. Added typeof guard.

File: packages/github/src/repoPattern.ts [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+ */
+export function parseRepoPatterns(config: string): RepoPattern[] {
+  const lines = config.split('\n');
+  const patterns: RepoPattern[] = [];
+
+  for (const rawLine of lines) {
+    // Strip inline comments (# preceded by whitespace) and trim
+    const line = rawLine.replace(/\s+#.*$/, '').trim();
+    if (line === '' || line.startsWith('#')) {
+      continue;
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts:20]: Fixed in ad710e3 — JSDoc now says 'trailing comments' not 'inline comments'.

File: packages/github/src/baseGithubProvider.ts [hunk 21]
```diff
@@ -96,4 +97,20 @@ export abstract class BaseGitHubProvider extends BaseProvider {
...
+  protected getConfiguredPatterns(): RepoPattern[] {
+    const config = vscode.workspace.getConfiguration('devDocketGithub');
+    const value = config.get<string | string[]>('repos', '');
+    if (!value || (Array.isArray(value) && value.length === 0)) { return []; }
+    // Backward compat: handle old array format
+    if (Array.isArray(value)) {
+      return parseRepoPatterns(value.join('\n'));
+    }
+    return parseRepoPatterns(value);
+  }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/baseGithubProvider.ts:109]: Fixed in 529d6c0 — removed legacy array compat (setting renamed to filteredRepos) and added typeof guard.

File: packages/github/src/githubMyPrsProvider.ts [hunk 22]
```diff
@@ -214,83 +160,18 @@ export class GitHubMyPrsProvider extends BaseGitHubProvider {
...
       logger.error('Failed to fetch authored PRs', err);
-      return { prs: [], failures: ['all repositories'] };
+      return { prs: [], failures: ['global fetch'] };
     }
 
     if (!response.ok) {
       logger.error(`Failed to fetch authored PRs: ${response.status}`);
-      return { prs: [], failures: ['all repositories'] };
+      return { prs: [], failures: ['global fetch'] };
     }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts:167]: Fixed in 30f3b72 — now uses 'authored PR search' as the failure identifier.

File: packages/github/src/githubMyPrsProvider.ts [hunk 23]
```diff
@@ -308,12 +189,12 @@ export class GitHubMyPrsProvider extends BaseGitHubProvider {
...
       logger.error('Failed to fetch assigned PRs', err);
-      return { prs: [], failures: ['all repositories'] };
+      return { prs: [], failures: ['global fetch'] };
     }
 
     if (!response.ok) {
       logger.error(`Failed to fetch assigned PRs: ${response.status}`);
-      return { prs: [], failures: ['all repositories'] };
+      return { prs: [], failures: ['global fetch'] };
     }
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts:196]: Fixed in 30f3b72 — now uses 'assigned PR search' as the failure identifier.

File: packages/github/src/baseGithubProvider.ts [hunk 24]
```diff
@@ -96,4 +97,15 @@ export abstract class BaseGitHubProvider extends BaseProvider {
...
   ): Promise<void>;
+
+  /**
+   * Read the `devDocketGithub.filteredRepos` setting and parse it into repo patterns.
+   */
+  protected getConfiguredPatterns(): RepoPattern[] {
+    const config = vscode.workspace.getConfiguration('devDocketGithub');
+    const value = config.get<string>('filteredRepos', '');
+    if (!value || typeof value !== 'string') { return []; }
+    return parseRepoPatterns(value);
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/baseGithubProvider.ts:108]: By design — the setting was renamed from repos to filteredRepos, so legacy string[] values won't appear under the new key. The typeof guard safely ignores unexpected types.

File: packages/github/src/test/githubProvider.test.ts [hunk 25]
```diff
@@ -58,48 +58,65 @@ describe('GitHubIssueProvider', () => {
...
+    });
+
+    const listener = vi.fn();
+    provider.onDidDiscoverItems(listener);
+    await provider.refresh();
+
+    expect(listener).toHaveBeenCalledTimes(1);
+    // Non-string value is ignored — no filtering applied, all items kept
+    const items = listener.mock.calls[0][0];
+    expect(items).toHaveLength(2);
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/test/githubProvider.test.ts:119]: Intentional — since the setting was renamed, array values under filteredRepos are unexpected. The test verifies they are safely ignored.

File: packages/github/src/repoPattern.ts [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+      isNegation = true;
+      patternText = patternText.slice(1).trim();
+    }
+
+    if (patternText === '') {
+      continue;
+    }
+
+    const regex = patternToRegex(patternText);
+    patterns.push({ pattern: line, isNegation, regex });
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts]: Fixed in 64371b4 — pattern field now stores normalized text consistent with the regex.

File: packages/github/src/repoPattern.ts [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+    if (patternText.startsWith('!')) {
+      isNegation = true;
+      patternText = patternText.slice(1).trim();
+    }
+
+    if (patternText === '') {
+      continue;
+    }
+
+    const regex = patternToRegex(patternText);
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/repoPattern.ts:39]: Fixed in 2474d90 — patterns without a / are now silently skipped with a test.

File: packages/github/src/githubPrReviewProvider.ts [hunk 28]
```diff
@@ -30,31 +31,42 @@ export class GitHubPrReviewProvider extends BaseGitHubProvider {
...
-    const repos = this.getConfiguredRepos();
-    const { prs, failures } = await this.fetchReviewRequestedPrs(accessToken, repos, signal);
+    const patterns = this.getConfiguredPatterns();
 
-    logger.info(`Discovered ${prs.length} PR review requests`);
+    const { prs, failed } = await this.fetchAllPrReviews(accessToken, signal);
+
+    // Parse repo name once per PR
+    const repoNameMap = new Map(prs.map(pr =>
+      [pr.html_url, parseRepoFromUrls(pr.html_url, pr.repository_url)]
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubPrReviewProvider.ts:40]: This is a pre-existing limitation of the global fetch approach (per_page=100), not introduced by patterns. Pagination for the Search API would be the proper fix but is out of scope for this PR.

File: packages/github/src/githubMyPrsProvider.ts [hunk 29]
```diff
@@ -40,12 +41,12 @@ export class GitHubMyPrsProvider extends BaseGitHubProvider {
...
   protected async fetchAndPublish(accessToken: string, isUserTriggered: boolean, signal?: AbortSignal): Promise<void> {
     logger.info('Fetching authored and assigned PRs...');
-    const repos = this.getConfiguredRepos();
+    const patterns = this.getConfiguredPatterns();
 
     // Fetch authored and assigned PRs in parallel; proceed with partial results if one fails
     const [authoredSettled, assignedSettled] = await Promise.allSettled([
-      this.fetchAuthoredPrs(accessToken, repos, signal),
-      this.fetchAssignedPrs(accessToken, repos, signal),
+      this.fetchAllAuthoredPrs(accessToken, signal),
```
[the remark this answers is not in the document]
Comment (mthalman) [reply packages/github/src/githubMyPrsProvider.ts:48]: Same as above — pre-existing Search API pagination limitation, not introduced by the pattern feature. Can be addressed in a follow-up.
